pkg/registry/core/service/storage/rest.go

```diff
@@ -810,3 +825,61 @@ func collectServiceNodePorts(service *api.Service) []int {
 	}
 	return servicePorts
 }
+
+// nearlyNullStrategy does almost nothing, but makes the REST logic happy.
```

Comment:
I'm in deewaters here, so is mostly for learning purposes, so
in Create we call
```
	if err := rest.BeforeCreate(rs.strategy, ctx, obj); err != nil {
```
with this `nearlyNullStrategy`, and here it can call `strategy.GenerateName()`
https://github.com/kubernetes/kubernetes/blob/2af52db68998440fe523d57d763d46307bfdcaab/staging/src/k8s.io/apiserver/pkg/registry/rest/create.go#L91-L93

but I don`t see that we implement it here, doesn't matter?
or should we mock it too?

Reply:
GenerateName is implemented by names.SimpleNameGenerator which we embed in nearlyNullStrategy in NewREST() (pkg/registry/core/service/storage/rest.go).  This was copied from the "real" strategy.